src/dataset-builder/DatasetBuilder.ts

```diff
@@ -665,23 +653,48 @@ export const DatasetBuilderContents = ({
       requestValid &&
         h(ActionBar, {
           prompt: h(Fragment, [
-            datasetRequestParticipantCount.status === 'Ready'
-              ? formatCount(datasetRequestParticipantCount.state.result.total)
+            snapshotRequestParticipantCount.status === 'Ready'
+              ? formatCount(snapshotRequestParticipantCount.state.result.total)
               : h(Spinner),
             ' participants in this dataset',
           ]),
-          actionText: 'Request access to this dataset',
-          onClick: () => setRequestingAccess(true),
+          actionText: 'Request this data snapshot',
+          onClick: () => {
+            setSnapshotAccessRequest(
+              withErrorReporting('Error creating dataset request')(
+                async () =>
+                  await DataRepo()
+                    .snapshotAccessRequest()
+                    .createSnapshotAccessRequest(
+                      createSnapshotAccessRequest(
+                        '',
+                        '',
```

Comment:
Do we want to file a ticket to remove research purpose from the SnapshotAccessRequest?

Reply:
yes!